GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -216,6 +216,35 @@ public virtual bool TryCreateToolsDirectory(out string upgraderToolPath, out str
             return false;
         }
 
+        public bool TryCleanup(out string error)
+        {
+            bool isError = false;
+            string exceptionInfo = string.Empty;
+            foreach (Asset asset in this.newestRelease.Assets)
+            {
+                Exception exception;
+                if (!this.TryDeleteDownloadedAsset(asset, out exception))
+                {
+                    exceptionInfo += $"Could not delete {asset.LocalPath}. {exception.ToString()}.";

```

Comment:
Add a newline at the end?

Reply:
UpgradeOrchestrator does that. It adds a newline when the message goes to Console, and Tracer adds the line when logging to file.

Reply:
But this code is in a loop here. If you have multiple exceptions, you're going to end up with a run-on string and no breaks between them.